### ISSUE TITLE
Net Core Conversion

### ISSUE THREAD

**Issue by nemesis567** [AlturosDestinations/Alturos.Yolo#16]
I've tried to port this to net core, but I have been unable to run it with this error:

Exception thrown: 'System.DllNotFoundException' in YoloCore.dll
Exception thrown: 'System.DllNotFoundException' in System.Private.CoreLib.dll
Exception thrown: 'System.DllNotFoundException' in System.Private.CoreLib.dll
Exception thrown: 'System.DllNotFoundException' in System.Private.CoreLib.dll
An unhandled exception of type 'System.DllNotFoundException' occurred in System.Private.CoreLib.dll
Unable to load DLL 'x64\yolo_cpp_dll_cpu.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E)

If you help me deal with this I can update a new branch here with net core.

**Comment by tinohager** [comment]
on which os you are test?

**Comment by nemesis567** [comment]
Windows 10 pro

**Comment by nemesis567** [comment]
No, it gives the same error with both of them.

**Comment by tinohager** [comment]
Can you check the c++ dll with http://www.dependencywalker.com/

**Comment by nemesis567** [comment]
Yes, I did it throws a bunch of errors and I can't see anything properly. I can use OpenCVSharp Yolo V2 with dnn.readNetFromDarknet

**Comment by tinohager** [comment]
Can you send a screenshot?

**Comment by Fusion86** [comment]
I started working on a .NET Standard port [here](https://github.com/Fusion86/Alturos.Yolo/tree/netcore).
It looks like you can't import DLL's from subfolders in .NET Standard.
My port seems to work fine on the CPU and the GPU, but it is still limited to Windows (for now) because all the included pre-compiled libraries (dependencies) are obviously only made for Windows.

**Comment by tinohager** [comment]
Can we set the target Framework to this?`<TargetFrameworks>net45;net461;netstandard2.0</TargetFrameworks>`

Yes my librariers are pre-compiled for windows. I need a solution to compile it for both systems. For example a docker container compile the librariers.

**Comment by Fusion86** [comment]
Is there a specific reason why you also want it to specify net461? Since [netstandard2.0 is already compatible with net461](https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support).

**Comment by tinohager** [comment]
yes if you install a .netstandard dll in a .net 4.6.1 project you have many extra dlls. I like less :)

**Comment by alexgurenko** [comment]
> I started working on a .NET Standard port [here](https://github.com/Fusion86/Alturos.Yolo/tree/netcore).
> It looks like you can't import DLL's from subfolders in .NET Standard.
> My port seems to work fine on the CPU and the GPU, but it is still limited to Windows (for now) because all the included pre-compiled libraries (dependencies) are obviously only made for Windows.

Hello! Did you manage to adapt your fork for Linux? I want to try running this on Jetson Nano )

**Comment by Fusion86** [comment]
No, I never got that far (since my most powerful PC is on windows I didn't really care). But if I remember correctly the the only thing (in theory) that needs to happen is to compile yolo_cpp_dll for linux. The only problem is that there is no makefile or anything like that for it, just a visual studio project. Now this is the point where I stopped caring about making it work for linux, so I don't know how hard it would be to compile for linux (maybe feeding the files to gcc works just fine, or maybe you need to edit the source a bit since it was made specifically for windows). I might try to look into this soon, but I wouldn't hold my breath.

**Comment by tinohager** [comment]
the yolo_cpp_dll can compile for linux check this [How to compile on Linux](https://github.com/AlexeyAB/darknet#how-to-compile-on-linux)

**Comment by tinohager** [comment]
I have no linux enviorment for testing i have only my windows system

**Comment by alexgurenko** [comment]
Thanks for advice,  @Fusion86, @tinohager  ! I already have yolo_console_dll.cpp  for linux-arm64. I will try it in near time
